mismatched types error when comparing aliased types

### The following program `sample.go` triggers an unexpected result

```go
package main

type Number = int

func main() {
	println(Number(1) < int(2))
}
```


### Expected result

```console
$ go run ./sample.go
true
```


### Got

```console
$ yaegi ./sample.go
run: ./sample.go:6:10: invalid operation: mismatched types main.Number and int
```


### Yaegi Version

v0.13.0

### Additional Notes

_No response_

--- a/_test/issue-1421.go
+++ b/_test/issue-1421.go
@@ -1,0 +1,9 @@
+package main
+
+type Number = int
+
+func main() {
+	println(Number(1) < int(2))
+}
+
+// Output: true

--- a/interp/typecheck.go
+++ b/interp/typecheck.go
@@ -192,7 +192,7 @@ func (check typecheck) comparison(n *node) error {
 
 	ok := false
 
-	if !isInterface(t0) && !isInterface(t1) && !t0.isNil() && !t1.isNil() && t0.untyped == t1.untyped && t0.id() != t1.id() {
+	if !isInterface(t0) && !isInterface(t1) && !t0.isNil() && !t1.isNil() && t0.untyped == t1.untyped && t0.id() != t1.id() && !typeDefined(t0, t1) {
 		// Non interface types must be really equals.
 		return n.cfgErrorf("invalid operation: mismatched types %s and %s", t0.id(), t1.id())
 	}

